orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -76,6 +83,62 @@ def _onnx_value_info_to_buffer_tensor(value_info, device):
     dtype = _utils.dtype_onnx_to_torch(value_info.type.tensor_type.elem_type)
     return torch.zeros(shape, device=device, dtype=dtype)
 
+def _parse_inputs_for_onnx_export(module, inputs):
+    # Ignore optional *inputs explicitly specified as None
+    sig = signature(module.forward)
+    all_input_names = sig.parameters.keys()
+    input_names = []
+    dynamic_axes = {}
+    for input_idx, name in enumerate(all_input_names):
+        if input_idx < len(inputs) and inputs[input_idx] is not None:
+            input_names.append(name)
+            dynamic_axes[name] = {}
+            for dim_idx in range(len(inputs[input_idx].shape)):
+                dynamic_axes[name].update({dim_idx : 'input{}_dim{}'.format(input_idx, dim_idx)})
+    return input_names, dynamic_axes
+
+def _parse_outputs_for_onnx_export(module, inputs):
+    #   Do an inference to grab outputs
+    is_train_mode = module.training
+    module.eval()
+    with torch.no_grad():
```

Comment:
May we restructure this code such that only things that need to be inside ```torch.no_grad()``` are inside this context?
If all of it is required inside the block, we could create another function for that piece so that it looks like:
```
output_names = []
output_dynamic_axes = {}
with torch.no_grad():
    _run_eval_and_parse_outputs()
if is_train_mode:
        module.train()
    return output_names, output_dynamic_axes
```

Reply:
The previous comment improved this logic. Keeping torch.no_grad for all the parsing of tensors doesn't bring perf hits nor logical consequences